griddedio/GriddedIO.F90

```diff
@@ -403,14 +413,18 @@ subroutine modifyTime(this, oClients, rc)
         type(StringVariableMap) :: var_map
         integer :: status
 
-        v = this%timeInfo%define_time_variable(rc=status)
-        _VERIFY(status)
-        call this%metadata%modify_variable('time',v,rc=status)
-        _VERIFY(status)
-        if (present(oClients)) then
-           call var_map%insert('time',v)
-           call oClients%modify_metadata(this%write_collection_id, var_map=var_map, rc=status)
+        if (this%timeInfo%is_initialized) then
+           v = this%timeInfo%define_time_variable(rc=status)
+           _VERIFY(status)
+           call this%metadata%modify_variable('time',v,rc=status)
            _VERIFY(status)
```

Comment:
I have some thoughts on the naming:  timeInfo%define_time_variable  from MAPL_TimeMethods.F90.   This subroutine shows:
```
subroutine modifyTime(this, oClients, rc)
           v = this%timeInfo%define_time_variable(rc=status)
           call this%metadata%modify_variable('time',v,rc=status)
```
Initially I thought  function define_time_variable  should appear only once. 
Then I see  the `define_time_variable`   in MAPL_TimeMethods.F90 did the following:
 - defines the time variable like long_name
 - read the clock to get the current time value to define begin_date, time_increment

I wonder if this should be called:  extract_time_variable_attribute_from_clock